Watchman.DomainModel/Warns/Commands/Handlers/RemoveWarnEventsCommandHandler.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Watchman.Cqrs;
+using Watchman.Integrations.MongoDB;
+
+namespace Watchman.DomainModel.Warns.Commands.Handlers
+{
+    public class RemoveWarnEventsCommandHandler : ICommandHandler<RemoveWarnEventsCommand>
+    {
+        private readonly ISessionFactory _sessionFactory;
+
+        public RemoveWarnEventsCommandHandler(ISessionFactory sessionFactory)
+        {
+            this._sessionFactory = sessionFactory;
+        }
+
+        public Task HandleAsync(RemoveWarnEventsCommand command)
+        {
+            if (command.GrantorId == null && command.ReceiverId == null)
+            {
+                return Task.CompletedTask;
+            }
+            using var session = this._sessionFactory.Create();
+            return session.DeleteAsync<WarnEvent>(x =>
+                    (command.GrantorId == null || command.GrantorId == x.GrantorId) &&
+                    (command.ReceiverId == null || command.ReceiverId == x.ReceiverId) &&
+                    (x.CreatedAt >= command.From) &&
+                    (x.ServerId == command.ServerId)
```

Comment:
ta kolejność jest celowo taka z myślą o wydajności
czy wyszło przypadkiem?

Reply:
przypadkiem, a to po kolei obcina znalezione wyniki według kolejności tutaj?

Reply:
chodzi o wydajność
najlepiej szukać po wartościach
od tych których jest najmniej do tych których jest najwięcej

żeby wykonać jak najmniej akcji sprawdzania

Reply:
```csharp
(x.ServerId == command.ServerId)
&& (command.ReceiverId == null || command.ReceiverId == x.ReceiverId)
&& (command.GrantorId == null || command.GrantorId == x.GrantorId)
&& (x.CreatedAt >= command.From)
```
Takie coś może być? Raczej częściej będą usuwane warny wg receiverId dlatego jest pierwsze

Reply:
myśle że jest ok

ewentualnie można przemyśleć wrzucenie sprawdzania daty na start
ale to kwestia tego jak będzie wyglądała tutaj sytuacja

czy dużo osób dostaje mało warnów
czy mało osób dostaje dużo warnów -  bo te osoby sie powtarzają

raczej dużo osób mało warnów
więc obecna wersja jest ok